### ISSUE TITLE
Error creating bean with name 'instancesProxyController' defined in class

### ISSUE THREAD

**Issue by bhagyaperera123** [spring-projects/spring-boot#15071]
I upgraded my spring boot application with below configuration.

Spring boot 2.0.4.RELEASE to 2.1.0.RELEASE
OpenJDK 1.8 to OpenJDK 11
Spring boot Admin version is 2.0.4

before that my application worked well. but now i have bean creation error. below i attached my stack trace.

few months ago i had this error. after that i found my issues. i had two beans with same name. 
i'm my project i have 'Instance Repository' interface and spring boot admin have same name bean.
then i added this annotation parameter in my project after that my project worked well. ( defaultInstanceRepository )

```
@Repository("defaultInstanceRepository")
public interface Instance Repository extends JpaRepository<Instance> {
}
```

but now i have this problem again after upgrade spring boot , spring boot admin and java. 

```
2018-11-02 17:49:10.209  WARN 14809 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'instancesProxyController' defined in class path resource [de/codecentric/boot/admin/server/config/AdminServerWebConfiguration$ServletRestApiConfirguation.class]: Unsatisfied dependency expressed through method 'instancesProxyController' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'instanceRegistry' defined in class path resource [de/codecentric/boot/admin/server/config/AdminServerAutoConfiguration.class]: Unsatisfied dependency expressed through method 'instanceRegistry' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'instanceRepository' defined in class path resource [de/codecentric/boot/admin/server/config/AdminServerAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [de.codecentric.boot.admin.server.domain.entities.SnapshottingInstanceRepository]: Factory method 'instanceRepository' threw exception; nested exception is java.lang.NoSuchMethodError: reactor.retry.Retry.retryMax(I)
```

**Comment by bhagyaperera123** [comment]
i know my explanation is not good. sorry about that.

**Comment by bhagyaperera123** [comment]
actually this error not related with my InstanceRepository interface. i created new project i added spring boot admin configuration and the i ran. i have same error with this project.
`2018-11-02 18:14:41.564 ERROR 16606 --- [           main] o.s.b.web.embedded.tomcat.TomcatStarter  : Error starting Tomcat context. Exception: org.springframework.beans.factory.BeanCreationException. Message: Error creating bean with name 'servletEndpointRegistrar' defined in class path resource [org/springframework/boot/actuate/autoconfigure/endpoint/web/ServletEndpointManagementContextConfiguration$WebMvcServletEndpointManagementContextConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.endpoint.web.ServletEndpointRegistrar]: Factory method 'servletEndpointRegistrar' threw exception; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'healthEndpoint' defined in class path resource [org/springframework/boot/actuate/autoconfigure/health/HealthEndpointConfiguration.class]: Unsatisfied dependency expressed through method 'healthEndpoint' parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'healthIndicatorRegistry' defined in class path resource [org/springframework/boot/actuate/autoconfigure/health/HealthIndicatorAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.health.HealthIndicatorRegistry]: Factory method 'healthIndicatorRegistry' threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.actuate.autoconfigure.jdbc.DataSourceHealthIndicatorAutoConfiguration': Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.actuate.autoconfigure.jdbc.DataSourceHealthIndicatorAutoConfiguration$$EnhancerBySpringCGLIB$$9b3f205d]: Constructor threw exception; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'dataSource' defined in class path resource [org/springframework/boot/autoconfigure/jdbc/DataSourceConfiguration$Hikari.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.zaxxer.hikari.HikariDataSource]: Factory method 'dataSource' threw exception; nested exception is org.springframework.boot.autoconfigure.jdbc.DataSourceProperties$DataSourceBeanCreationException: Failed to determine a suitable driver class
2018-11-02 18:14:41.576  INFO 16606 --- [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]`

**Comment by joshiste** [comment]
We are going to release a new version of SBA next week which doesn't suffer from this issue until then you can use on of our snapshot versions.

**Comment by wilkinsona** [comment]
Thank you, @joshiste. Sorry for the trouble caused by the binary incompatible change. Was it intentional, @simonbasle? Is there some documentation that we could link to in Boot’s release notes?

**Comment by simonbasle** [comment]
@wilkinsona yes there was a deliberate decision to make this one of the (thankfully few) binary breaking changes in 3.2.0 (or rather `Californium` release train's first GA).

It is documented in the [release notes](https://github.com/reactor/reactor-addons/releases/tag/v3.2.0.RELEASE), the rationale being that having an `int` there effectively limits the possible number of retries by a significant amount compared to a `long` (and also, core uses `long`).
